packages/flutter_tools/test/base/build_test.dart

```diff
@@ -0,0 +1,192 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:file/memory.dart';
+import 'package:flutter_tools/src/base/build.dart';
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:test/test.dart';
+
+import '../src/context.dart';
+
+/// Matcher for functions that throw [BuildException].
+final Matcher throwsBuildException = throwsA(const isInstanceOf<BuildException>());
+
+void main() {
+  group('BuildAction', () {
```

Comment:
Can you set `fs` to a memory file system for all tests so these are are hermetic?